lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2554,8 +2554,15 @@ namespace {
         if (operationFn->getParameters()->size() != 1)
           return false;
 
-        if (operationFn->getParameters()->get(0)->getInterfaceType()->isEqual(
-                classDecl->getInterfaceType()))
+        auto paramType =
+            operationFn->getParameters()->get(0)->getInterfaceType();
+        if (auto optionalType = paramType->getOptionalObjectType()) {
+          paramType = optionalType;
+        }
+        auto paramDecl = paramType->getAnyNominal();
```

Comment:
Is `paramDecl` a pointer? We use `auto *` for pointers, it might also be clearer to spell out the type if it is not too verbose: https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable.  It would also be better to spell out the type of `paramType` (which is Type?).

Reply:
Swift doesn't use the LLVM code style though. One of the most noticeable divergences is probably that methods in the Swift codebase start from a lower-case letter, but in LLVM they are capitalized.